### DISCUSSION TITLE
Fail to build docker commit: 9db3d792cc58e0fec321a26fc3da327e428d369a

### DISCUSSION THREAD

**Discussion by Originalimoc** [danny-avila/LibreChat#1329] [Troubleshooting] [answered]
### What happened?

docker-compose build: WARN BADENGINE
docker-compose up: Error: Cannot find module 'langchain/schema'

### Steps to Reproduce

git pull, docker-compose build.
vite require "^18.0.0 | >=20.0.0" npm but installed 19.9

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I can't reproduce it. Try the following:
```bash
# Stop the container (if running)
docker-compose down
# Fetch the latest changes from Github
git fetch origin
# Switch to the repo's main branch
git checkout main
# Pull the latest changes to the main branch from Github
git pull origin main
# Prune all LibreChat Docker images
docker rmi librechat:latest
# Remove all unused dangling Docker images
docker image prune -f
# Building a new LibreChat image
docker-compose build

# Start LibreChat
docker-compose up
```

**Reply by danny-avila** [reply]
If you still have issues, try the following:

Install npm if you don't have it already:
>Navigate to https://nodejs.org/en/download and to download the latest Node.js version for your OS (The Node.js installer includes the NPM package manager.)

after it's installed and you're in the LibreChat directory, run `npm run update:docker` and please share your logs.

Also please share your docker version, Operating System version as well.

**Comment by danny-avila** [comment]
I'm in the middle of updating a computer that was running on a commit dating back to June (last PR around #530). Will report back as it runs slow to update.

**Comment by danny-avila** [comment, marked as answer]
> WARN BADENGINE

Nothing to worry about here. Vite is pushing this warning because we are on node v19 and it has reach its End of life, as it will no longer be maintained; therefore, vite will not raise concern over issues using this engine. This is very recent, and we will move to node:20-alpine instead of 19 soon once I can debug other unrelated issues related to this upgrade with the default compose file.

> Error: Cannot find module 'langchain/schema'

This indicates to me that the image is out of sync as is the same issue with the other discussion's issue `Error: Cannot find module '/app/api/server/index.js'` (https://github.com/danny-avila/LibreChat/discussions/776)

 I was able to reproduce the issue on a computer on an old commit and it's an issue with the compose file, should be fixed with this: https://github.com/danny-avila/LibreChat/pull/1330

**Comment by ducphamhoang** [comment]
Same here. I tried "npm run update:docker" and here is the logs:
```
E:\ChatAI\LibreChat>npm run update:docker

> LibreChat@0.6.1 update:docker
> node config/update.js -d

Starting update script, this may take a minute or two depending on your system and network.
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
Fetching the latest repo...
Switching to main branch...
Already on 'main'
D       .env.example
M       docker-compose.yml
Your branch is behind 'origin/main' by 3 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Pulling the latest code from main...
From https://github.com/danny-avila/LibreChat
 * branch              main       -> FETCH_HEAD
error: Your local changes to the following files would be overwritten by merge:
        docker-compose.yml
Please commit your changes or stash them before you merge.
Aborting
Updating 9db3d792..6829d66c
node:internal/errors:932
  const err = new Error(message);
              ^

Error: Command failed: git pull origin main
    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at E:\ChatAI\LibreChat\config\update.js:77:5
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 18592,
  stdout: null,
  stderr: null
}

Node.js v20.10.0.
```

Tried as you suggest:
 ```
"# Stop the container (if running)
docker-compose down
# Fetch the latest changes from Github
git fetch origin
# Switch to the repo's main branch
git checkout main
# Pull the latest changes to the main branch from Github
git pull origin main
# Prune all LibreChat Docker images
docker rmi librechat:latest
# Remove all unused dangling Docker images
docker image prune -f
# Building a new LibreChat image
docker-compose build

# Start LibreChat
docker-compose up" 
```

still no luck

**Reply by danny-avila** [reply]
@ducphamhoang this is a different issue for you. As you can see, git failed for you and you made local changes to the compose file. Stash the changes (look into how to use git stash) then try again

**Reply by danny-avila** [reply]
You can either remove your changes or use as follows

To manage your changes in `./docker-compose.yml` and update your repository, you'll need to use Git commands. Here's how you can do it:

1. **Stash Your Changes**: Save your current changes without committing them.
   ```bash
   git stash push -m "Your stash message" ./docker-compose.yml
   ```
   This command stashes the changes made to `./docker-compose.yml` and keeps your working directory clean. The `-m` option allows you to add a custom message to your stash for easier identification later.

2. **Pull the Latest Changes**: update with npm run update as before

3. **Reapply Your Stashed Changes**: Once the pull is complete, you can reapply the changes you stashed earlier.
   ```bash
   git stash pop
   ```
   This command applies the most recently stashed changes to your working directory and removes them from the stash list. If you want to keep the changes in the stash list, use `git stash apply` instead.

Let me know if you’re still having issues

**Reply by ducphamhoang** [reply]
Yes thank @danny-avila , that's git issue that i completely missed before. It works now without any problem.
